core/trino-main/src/main/java/io/trino/server/security/oauth2/RefreshTokensConfig.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server.security.oauth2;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.ConfigSecuritySensitive;
+import io.airlift.units.Duration;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+
+import javax.crypto.SecretKey;
+import javax.validation.constraints.NotEmpty;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.util.concurrent.TimeUnit.HOURS;
+
+public class RefreshTokensConfig
+{
+    private Duration refreshTokenExpiration = Duration.succinctDuration(1, HOURS);
+    private String issuer = "Trino_coordinator";
+    private String audience = "Trino_coordinator";
+    private SecretKey secretKey;
+
+    public Duration getRefreshTokenExpiration()
+    {
+        return refreshTokenExpiration;
+    }
+
+    @Config("http-server.authentication.oauth2.refresh-tokens.refresh.timeout")
```

Comment:
I'm not sure about the description on overall naming convention. Using `refresh` and `refresh-tokens` for different things might be confusing. `Expiration time for refresh token` it's not really expiration time of the refresh token but the token issued by Trino.